crates/viewer/re_viewer_context/src/test_context.rs

```diff
@@ -266,17 +302,30 @@ impl TestContext {
             selection_state: &self.selection_state,
             blueprint_query: &self.blueprint_query,
             egui_ctx,
-            render_ctx: render_ctx.as_deref(),
+            render_ctx: Some(render_ctx),
             command_sender: &self.command_sender,
             focused_item: &None,
             drag_and_drop_manager: &drag_and_drop_manager,
         };
 
         func(&ctx);
 
-        if let Some(render_ctx) = render_ctx {
-            render_ctx.before_submit();
+        // If re_renderer was used, `setup_kittest_for_rendering` should have been called.
+        // (if not, then we clearly didn't render anything since that requires writing to a texture!).
+        if render_ctx
+            .active_frame
+            .num_view_builders_created
+            .load(std::sync::atomic::Ordering::Acquire)
+            != 0
+            && !self
+                .called_setup_kittest_for_rendering
+                .load(std::sync::atomic::Ordering::Relaxed)
```

Comment:
This is very hard to read.
```suggestion
        let num_view_builders_created =
            render_ctx.active_frame.num_view_builders_created();
        let called_setup_kittest_for_rendering =
            self.called_setup_kittest_for_rendering.load(Ordering::Relaxed);
        if num_view_builders_created != 0 && !called_setup_kittest_for_rendering
```

Reply:
Tip: RustAnalyzer has helpers for extracting a selected expression into a named variable